pkg/operator/ceph/config/monstore.go

```diff
@@ -207,3 +191,64 @@ func (m *MonStore) SetKeyValue(key, value string) error {
 	}
 	return nil
 }
+
+func (m *MonStore) SetAll(clientName string, settings map[string]string) error {
+	assimilateConfPath, err := ioutil.TempFile("", "")
+	if err != nil {
+		return errors.Wrapf(err, "failed to create assimilateConf temp dir for  %s.", clientName)
+	}
+
+	err = ioutil.WriteFile(assimilateConfPath.Name(), []byte(""), 0600)
+	if err != nil {
+		rook.TerminateFatal(errors.Wrapf(err, "failed to write config file"))
+	}
+
+	defer func() {
+		err := os.Remove(assimilateConfPath.Name())
+		if err != nil {
+			logger.Errorf("failed to remove file %q. %v", assimilateConfPath.Name(), err)
```

Comment:
should this error be returned?

Reply:
We can't return an err from the defer function, otherwise agreed